Added min and max functions

joey2031 opened the pull request:
Fixes issue #7 at first Array.From did not work it gave me this error:
Property 'from' does not exist on type 'ArrayConstructor'. Do you need to change your target library? Try changing the `lib` compiler option to 'es2015' or later.ts(2550)
So I changed the target in esconfig.json. Let me know what you think! Thanks

joey2031 committed:
Added min and max functions

joey2031 committed:
Fixed conflicts

joey2031 committed:
Merge branch 'main' into issue-7

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -152,6 +152,16 @@ export default class AdvancedSet<T> {
     return this.toArray().every(test);
   }
 
+  public maxValue(setB: AdvancedSet<number>): number {
```
Instead of finding the min of another set, could you find the min of the current set? So using `this.XYZ` instead of taking in a parameter

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -152,6 +152,16 @@ export default class AdvancedSet<T> {
   }
 
+  public maxValue(setB: AdvancedSet<number>): number {
+    let arrSet = Array.from(setB);
```
Could you change this to use the built in `.toArray()` instead?

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -152,6 +152,16 @@ export default class AdvancedSet<T> {
+  }
+
+  public minValue(setB: AdvancedSet<number>): number {
+    let arrSet = Array.from(setB);
```
Same comments as in Max

jpbulman commented on tsconfig.json:
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es2015",
```
Could we keep the target es5, for backwards compatiblity?

jpbulman reviewed (changes requested):
Looks good, just left a couple of comments, thanks for starting this! Could you also add a couple tests for each of these?

joey2031 committed:
Changed target back to es5

joey2031 committed:
Added max and min function

joey2031 committed:
Added tests for max and min function

joey2031 commented:
@jpbulman Fixed everything, the command line was giving me a hard time so I just dragged and dropped the files I fixed, I think everything is correct. If you need me to fix anything let me know!

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -1,157 +1,166 @@
+    return new AdvancedSet(...this.toArray(), ...setB.toArray());
+  }
+
+  public difference(setB: AdvancedSet<T>): AdvancedSet<T> {
```
It looks like there were some formatting issues, as the diff seems to be the entire file. Could you try and copy/pasting your changes somewhere locally, checkout the file from `main`, and then add the changes back in?

joey2031 replied:
ill make a new PR to fix this

jpbulman commented on tsconfig.json:
```diff
@@ -1,12 +1,12 @@
-}
+{
+  "compilerOptions": {
+    "target": "es5",
```
Same formatting issues here

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -1,157 +1,166 @@
+  }
+
+  public maxValue(): number {
+    let arrSet = (this.toArray() as unknown[]) as number[];
```
Some of the typing here seems funky, why are you casting this twice?

joey2031 replied:
@jpbulman  Right, I originally had this 
![image](https://user-images.githubusercontent.com/36198917/104230802-0b47bb80-541c-11eb-9356-628e1e8b74a2.png)
But it gave this error 
Conversion of type 'T[]' to type 'number[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'T' is not comparable to type 'number'.ts(2352)
When I did it like let arrSet = (this.toArray() as unknown[]) as number[]; the error went away

jpbulman commented on src/AdvancedSet.ts:
```diff
@@ -1,157 +1,166 @@
+    return Math.max(...arrSet);
+  }
+
+  public minValue(): number {
```
Could you maybe just call the functions `min` and `max`?
